common/util/file_util.cc

```diff
@@ -46,6 +48,39 @@ absl::string_view Stem(absl::string_view filename) {
              : filename.substr(0, last_dot_pos);
 }
 
+bool UpwardFileSearch(absl::string_view start, absl::string_view filename,
+                      std::string& result) {
+  /* Convert to absolute path */
+  char absolute_path[PATH_MAX];
+  if (realpath(std::string(start).c_str(), absolute_path) < 0) {
+    return false;
+  }
+
+  /* Add trailing slash */
+  std::string init = absl::StrCat(absolute_path, "/");
+
+  size_t r = 0, n = init.length();
+  do {
+    r = init.rfind("/", n);
+
```

Comment:
In this loop, it would be a good idea to add some VLOG(1) statements to be able to trace this file access behavior for debugging which file was actually found.

Reply:
Added